controller/internal/enforcer/envoyauthorizer/envoyproxy/auth_server.go

```
@@ -0,0 +1,466 @@
+
+		if source != nil {
+			if addr := source.GetAddress(); addr != nil {
+				if sockAddr := addr.GetSocketAddress(); sockAddr != nil {
+					sourceIP = sockAddr.GetAddress()
+					srcPort = int(sockAddr.GetPortValue())
+				}
+			}
+		}
+		if dest != nil {
+			if destAddr := dest.GetAddress(); destAddr != nil {
+				if destSockAddr := destAddr.GetSocketAddress(); destSockAddr != nil {
+					destIP = destSockAddr.GetAddress()
+					destPort = int(destSockAddr.GetPortValue())
+				}
+			}
+		}
+
+		if request := attrs.GetRequest(); request != nil {
+			httpReq = request.GetHttp()
+			if httpReq != nil {
+				httpReqHeaders := httpReq.GetHeaders()
+				aporetoAuth, _ = httpReqHeaders[aporetoAuthHeader]
+				aporetoKey, _ = httpReqHeaders[aporetoKeyHeader]
+				zap.L().Debug("ext_authz ingress: ", zap.Any("httpReqHeaders: ", httpReqHeaders), zap.String("aporetoKey: ", aporetoKey))
+				urlStr = httpReq.GetPath()
+				method = httpReq.GetMethod()
+				scheme = httpReq.GetScheme()
+
+			}
+		}
+	}
+	zap.L().Debug("ext_authz ingress:", zap.String("source addr: ", sourceIP), zap.String("source, dest: ", source.GetAddress().GetSocketAddress().GetAddress()), zap.String("dest addr: ", dest.GetAddress().GetSocketAddress().GetAddress()))
+	zap.L().Debug("ext_authz ingress:", zap.Any("destPort: ", destPort), zap.Any("srcPort: ", srcPort), zap.String("scheme: ", scheme))
+
+	requestCookie := &http.Cookie{Name: aporetoAuthHeader, Value: aporetoAuth} // nolint errcheck
+	hdr := make(http.Header)
+
+	zap.L().Debug("ext_authz ingress:", zap.String("Aporeto-Auth: ", aporetoAuth), zap.String("Aporeto-key: ", aporetoKey))
+	hdr.Add(aporetoAuthHeader, aporetoAuth) //string(p.secrets.TransmittedKey()))
+	hdr.Add(aporetoKeyHeader, aporetoKey)   //resp.Token)
+
+	// Create the new target URL based on the method+path parameter that we had.
+	URL, err := url.ParseRequestURI("http:" + method + urlStr)
+	if err != nil {
+		fmt.Println("Cannot parse the URI", err)
+		return nil, nil
```

should return the error?